shpc/main/modules/base.py

```diff
@@ -269,18 +264,17 @@ def _cleanup_symlink(self, module_dir):
         symlinked_module = self.get_symlink_path(module_dir)
         if not symlinked_module:
             return
-        if os.path.exists(symlinked_module) and os.path.islink(symlinked_module):
+        if os.path.islink(symlinked_module):
```

Comment:
Huh - could we use the same logic here that we use for cleaning up module / container directories, except provide a function (to unlink) to overwrite the delete? There is a bit of duplicated logic.

Reply:
Absolutely. I almost committed the change in `utils.rmdir_to_base` but decided to it this way instead in order to minimise the number of files changed.
But you're right, `utils.rmdir_to_base` can be made to call `shutil.rmtree` on directories and `os.path.unlink` on symlinks. It'd be clean and reusable.

Reply:
https://github.com/singularityhub/singularity-hpc/pull/539/commits/61996aea11d1aabb1caef92b11384ef42a6679b2